### PR TITLE
Allow movefocus for empty workspaces

### PR DESCRIPTION
**Pull request by happenslol**
#### Describe your PR, what does it fix/add?

As discussed in #1999, this PR implements moving to and from empty workspaces using `movefocus`.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

I've tried to adapt to your naming schemes, but I'm not sure in which cases you use the `P` prefix for variables, and if I'm correct in assuming that all local window/workspace/monitor variables in functions should be all uppercase (e.g. `PNEWWINDOW`). I'd like having at least underscore to separate words for readability (`P_NEW_WINDOW`), but I'll use the current style if that's what you prefer.

I was also wondering, if a workspace has no current focused window, is it possible that there are still windows? I imagine that could happen if a silent move to a workspace occurred. See my TODO in the code for that.

#### Is it ready for merging, or does it need work?

I was unsure about how this functionality should be activated, since it's potentially breaking the workflow of current users who depend on a fallback to any window if no window is found in the direction. It would also be possible to still do that fallback behavior if no monitor was found in that direction.

My idea would be to add an option to enable the fallback behavior (`getNextWindowOnWorkspace`) and enable it by default.

### CODE REVIEW

File: src/managers/KeybindManager.cpp [hunk 1]
```diff
@@ -1153,14 +1181,38 @@ void CKeybindManager::moveFocusTo(std::string args) {
...
+            Vector2D middle = PNEWMONITOR->vecPosition + PNEWMONITOR->vecSize / 2.f;
+            g_pCompositor->warpCursorTo(middle);
+
+            return;
         }
     }
+
+    Debug::log(LOG, "No monitor found in direction %c", arg);
+
+    // TODO: Add option to disable looping/falling back to any next window?
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:1211]: I'd rather not yeet it atm if we don't have the option

File: src/managers/KeybindManager.cpp [hunk 2]
```diff
@@ -1153,14 +1181,38 @@ void CKeybindManager::moveFocusTo(std::string args) {
...
-        if (PWINDOWNEXT) {
-            switchToWindow(PWINDOWNEXT);
+        return;
+    }
+
+    Debug::log(LOG, "No window found in direction %c, looking for a monitor", arg);
+
+    // No window found in direction, switch to workspace in direction
+    const auto PNEWMONITOR = g_pCompositor->getMonitorInDirection(arg);
+    if (PNEWMONITOR && PNEWMONITOR->activeWorkspace != -1) {
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:1194]: redundant `activeWorkspace` check. it's guaranteed

File: src/managers/KeybindManager.cpp [hunk 3]
```diff
@@ -1113,9 +1113,37 @@ void CKeybindManager::moveFocusTo(std::string args) {
...
     const auto PLASTWINDOW = g_pCompositor->m_pLastWindow;
+    if (!PLASTWINDOW) {
+        // Try to move to monitor in direction
+        const auto PNEWMONITOR = g_pCompositor->getMonitorInDirection(arg);
+        if (!PNEWMONITOR) {
+            return;
+        }
+
+        const auto PNEWWORKSPACE = g_pCompositor->getWorkspaceByID(PNEWMONITOR->activeWorkspace);
+        if (!PNEWWORKSPACE) {
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:1124]: redundant

File: src/managers/KeybindManager.cpp [hunk 4]
```diff
@@ -1153,14 +1181,38 @@ void CKeybindManager::moveFocusTo(std::string args) {
...
+    }
+
+    Debug::log(LOG, "No window found in direction %c, looking for a monitor", arg);
+
+    // No window found in direction, switch to workspace in direction
+    const auto PNEWMONITOR = g_pCompositor->getMonitorInDirection(arg);
+    if (PNEWMONITOR && PNEWMONITOR->activeWorkspace != -1) {
+        const auto PNEWWORKSPACE = g_pCompositor->getWorkspaceByID(PNEWMONITOR->activeWorkspace);
+        if (PNEWWORKSPACE) {
+            g_pCompositor->focusWindow(nullptr);
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:1197]: what if there is a floating window? You should do the same as above, getlastfocusedwindow

**Review by vaxerski — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/43317083/230892463-25776994-3bcb-4f7a-a5cd-976f7dd8357c.mp4

File: src/managers/KeybindManager.cpp (hunk 1 above)
Comment (happenslol) [reply src/managers/KeybindManager.cpp:1211]: Can I just add an option? Cause I personally don't want that at all, so while I'm at it

**Comment by happenslol** [comment]
Ok i pull up

All except the option thingie should be fixed, tell me if you're ok with the helper function like that

File: src/managers/KeybindManager.cpp (hunk 1 above)
Comment (vaxerski) [reply src/managers/KeybindManager.cpp]: sure

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**